capture/capture_aes.py

```diff
@@ -0,0 +1,380 @@
+        pll_frequency = cfg["target"]["pll_frequency"],
+    )
+    scope = Scope(scope_cfg)
+
+    # Init project.
+    project_cfg = ProjectConfig(type = cfg["capture"]["trace_db"],
+                                path = project,
+                                wave_dtype = np.uint16,
+                                overwrite = True,
+                                trace_threshold = cfg["capture"].get("trace_threshold")
+                                )
+    project = SCAProject(project_cfg)
+    project.create_project()
+
+    return target, scope, project
+
+
+def configure_cipher(cfg, target, capture_cfg) -> OTAES:
+    """ Configure the AES cipher.
+
+    Establish communication with the AES cipher and configure the seed.
+
+    Args:
+        cfg: The project config.
+        target: The OT target.
+        capture_cfg: The capture config.
+
+    Returns:
+        The communication interface to the AES cipher.
+    """
+    # Create communication interface to OT AES.
+    ot_aes = OTAES(target.target)
+
+    # If batch mode, configure PRNGs.
+    if capture_cfg.batch_mode:
+        # Seed host's PRNG.
+        random.seed(cfg["test"]["batch_prng_seed"])
+
+        # Seed the target's PRNGs for initial key masking, and additionally
+        # turn off masking when '0'.
+        ot_aes.write_lfsr_seed(cfg["test"]["lfsr_seed"].to_bytes(4, "little"))
+        ot_aes.write_batch_prng_seed(cfg["test"]["batch_prng_seed"].to_bytes(4, "little"))
+
+    return ot_aes
+
+
+def generate_ref_crypto(sample_fixed, mode, key, key_fixed, plaintext, key_length):
+    """ Generate cipher material for the encryption.
+
+    This function derives the next key as well as the plaintext for the next
+    encryption.
+
+    Args:
+        sample_fixed: Use fixed key or new key.
+        mode: The mode of the capture.
+        key: The current key.
+        key_fixed: The fixed key for FVSR.
+        plaintext: The current plaintext.
+        key_length: Th length of the key.
+
+    Returns:
+        plaintext: The next plaintext.
+        key: The next key.
+        ciphertext: The next ciphertext.
+        sample_fixed: Is the next sample fixed or not?
+    """
+    if mode == "aes_random":
+        cipher = AES.new(bytes(key), AES.MODE_ECB)
+        ciphertext = bytearray(cipher.encrypt(bytes(plaintext)))
+    else:
+        if sample_fixed:
+            # Use fixed_key as this key.
+            key = np.asarray(key_fixed)
+        else:
+            # Generate this key from the PRNG.
+            key = bytearray(key_length)
+            for i in range(0, key_length):
+                key[i] = random.randint(0, 255)
+        # Always generate this plaintext from PRNG (including very first one).
+        plaintext = bytearray(16)
+        for i in range(0, 16):
+            plaintext[i] = random.randint(0, 255)
+        # Compute ciphertext for this key and plaintext.
+        cipher = AES.new(bytes(key), AES.MODE_ECB)
```

Comment:
The old capture script uses the scared.aes library for this purpose, as shown [here](https://github.com/lowRISC/ot-sca/blob/ce04758ecacc08ea9d55760e24dab8d5de817af9/cw/capture.py#L560). However, when I use this library instead of `AES.new(bytes(key), AES.MODE_ECB)`, the capture rate significantly drops. So I keep it for now and and add a TODO if we encounter problems with the capture rate.

Reply:
Ah, this is perfect then. Thanks for investigating!